website/content/docs/job-specification/consul.mdx

```diff
@@ -0,0 +1,236 @@
+description: |-
+  The "consul" block allows the group or task to specify that it requires a token
+  from a HashiCorp Consul server. Nomad will automatically retrieve a Consul token
+  for the group or task.
+---
+
+# `consul` Block
+
+<Placement
+  groups={[
+    ['job', 'group', 'consul'],
+    ['job', 'group', 'task', 'consul'],
+  ]}
+/>
+
+The `consul` block specifies [Consul][] configuration options specific to a
+task. If specified at the `group` level, the configuration will apply to all
+tasks and services within the group unless a task has its own `consul` block.
+
+```hcl
+job "docs" {
+  group "example" {
+    task "server" {
+      consul {
+        cluster   = "default"
+        namespace = "default"
+      }
+    }
+  }
+}
+```
+
+## Workload Identity
+
+Starting in Nomad 1.7, Nomad clients will use a task or service's [Workload
+Identity][] to authenticate to Consul and obtain a Consul token specific to the
+service or task when the following conditions are met:
+
+* For services, if either the [`consul.service_identity`][] is configured on the
+  Nomad servers or the service includes an `identity` block.
+* For tasks, if the [`consul.task_identity`][] is configured on the Nomad
+  servers, or the task includes an [`identity`][] block with the `name` field
+  set to `consul_default` (or `consul_$clusterName` for non-default Consul
+  clusters in Nomad Enterprise).
+
+As a fallback if none of these conditions are met, the Nomad client will instead
+use, in order of preference:
+
+* The token provided by the [`-consul-token`][flag_consul_token][] flag for the
+  `nomad job run` command.
+* The Consul token as configured in the agent's [`consul.token`][] configuration.
+* The Consul token as configured by the agent's `CONSUL_HTTP_TOKEN` environment variable.
+
+To avoid failed deployments, you should first set up an authentication method
+and binding rules in Consul before configuring the Nomad servers with
+`consul.task_identity` and `consul.service_identity`. Refer to [Configuring Consul
+Authentication][] for more details.
+
+<Warning>
+Starting in Nomad 1.9, the fallback options to use the -consul-token flag when
+submitting a job, the agent's consul.token configuration, or the
+CONSUL_HTTP_TOKEN environment variable, will be removed.  This means service and
+template blocks will not be able to use the agent's Consul token or one provided
+by the job submitter. You should be prepared to migrate to the Workload Identity
+workflow for Consul and Vault before upgrading to Nomad 1.9. Refer to <a
+href="/nomad/docs/integrations/consul-integration#migrating-to-using-workload-identity-with-consul">Migrating
+to Using Workload Identity with Consul</a>
+</Warning>
+
+### Access to Token
+
+The Nomad client will make the Consul token available to the task by writing it
+to the secret directory at `secrets/consul_token` and by injecting a
+`CONSUL_TOKEN` environment variable in the task. If the Nomad cluster is
```

Comment:
Adding `env` and `file` to the `consul` block (like `vault`) could be a nice future work.